Automattic/pocket-casts-ios
# Referrals: deep link for claiming guest pass

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

Update the referrals code to support deep links and proper check your current subscription status to see if it should show the claim screen or show you `The offer is not available screen`

At the moment we are still not validating the redeem code on the server so all `/redeem/*` urls will be valid.

| 1 | 2 | 3 |
| - | - | - |
| ![Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 52](https://github.com/user-attachments/assets/b010d3f8-c29b-4a6e-b9a8-aae66b5cdb3e) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 25](https://github.com/user-attachments/assets/b7dc199b-cc73-4d5b-adc0-c06bf93ad4b9) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-09-20 at 19 06 23](https://github.com/user-attachments/assets/c3b64a93-c3c2-4af7-980c-789978363fa4) |

## To test

1. Start the app
2. Ensure that you have the referrals FF enabled
3. Switch the app to the background
4. Open a redeem deep link 
    - **If testing on the simulator**: Open the the terminal and write this: `xcrun simctl openurl booted 'http://pocketcasts.com/redeem/test'` . You need to have only one simulator open 
    - If testing on the device: Send an email to yourself with the following link in the content: `http://pocketcasts.com/redeem/test`.  Open the email on the device and tap on the link
5. Check that  the app opens one of the follow:
   - If you don't have a subscription you see the Claim Guest pass screen 
   - If you already have an existing subscription you should see the `You already have a subscription screen`
6. Remember test both scenarios on Settings -> Developer you can change your subscription status 
7. If you saw the Claim guest pass screen, if you tap on `Not now` you should screen a cell on the profile allowing to go back to the claim screen.

## Checklist
